pkg/v1/mutate/mutate_test.go

```diff
@@ -266,6 +266,31 @@ func TestMutateConfig(t *testing.T) {
 	}
 }
 
+func TestMutateManifestAnnotations(t *testing.T) {
+	source := sourceImage(t)
+
+	newAnnotations := map[string]string{
+		"im.the.first.annotation": "hello world",
+	}
+
+	result, err := mutate.ManifestAnnotations(source, newAnnotations)
+	if err != nil {
+		t.Fatalf("failed to mutate a config: %v", err)
+	}
+
+	if manifestsAreEqual(t, source, result) {
+		t.Error("mutating the manifest annotations MUST mutate the manifest")
+	}
+
+	if configDigestsAreEqual(t, source, result) {
+		t.Errorf("mutating the manifest annotations MUST mutate the config digest")
```

Comment:
This actually should prove that the config isn't modified while modifying annotations.

Reply:
yes you are right, thank you so much for the feedbacks 🙏